parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java

```diff
@@ -217,6 +219,10 @@ public <T> Boolean lt(BoundReference<T> ref, Literal<T> lit) {
           return ROWS_CANNOT_MATCH;
         }
 
+        if (fieldPromotedToString(ref, colStats)) {
```

Comment:
the method name `fieldPromotedToString` is not great since technically it's not 100% that it really was promoted, we're using properties of the bounds to just assume that a field was promoted